src/functions.php

```diff
@@ -31,6 +46,14 @@ function reject($promiseOrValue = null)
         });
     }
 
+    if ($promiseOrValue instanceof AsyncInteropPromise) {
+        return new Promise(function ($resolve, $reject) use ($promiseOrValue) {
+            $promiseOrValue->when(function ($reason = null, $value = null) use ($resolve, $reject) {
+                $reject($reason ? $reason : $value);
```

Comment:
I just wasn't sure because of `UnhandledRejectionException::nullOrResolve`. `RejectedPromise` seems to miss a check for `AsyncInteropPromise` in `__construct` btw.